lib/src/loader.rs

```diff
@@ -105,19 +106,51 @@ fn load_elf(fd: &mut File, name: String) -> Result<(Project,Machine)> {
 
     prog.call_graph.add_vertex(CallTarget::Todo(Rvalue::new_u64(entry as u64),Some(name),Uuid::new_v4()));
 
-    let dynstrtab = binary.dynstrtab;
-    for sym in binary.dynsyms {
-        let name = dynstrtab[sym.st_name() as usize].to_string();
-        let addr = sym.st_value();
-        debug!("{} @ 0x{:x}: {:?}", name, addr, sym);
+    let add_sym = |prog: &mut Program, sym: &elf::Sym, strtab: &goblin::strtab::Strtab| {
+        let name = strtab[sym.st_name].to_string();
+        let addr = sym.st_value;
+        debug!("Symbol: {} @ 0x{:x}: {:?}", name, addr, sym);
         if sym.is_function() {
             if sym.is_import() {
                 prog.call_graph.add_vertex(CallTarget::Symbolic(name,Uuid::new_v4()));
             } else {
                 prog.call_graph.add_vertex(CallTarget::Todo(Rvalue::new_u64(addr),Some(name),Uuid::new_v4()));
             }
         }
+    };
+
+    let resolve_import_address = |proj: &mut Project, relocs: &[elf::Reloc], name: &str| {
+        for reloc in relocs {
+            let pltsym = &binary.dynsyms[reloc.r_sym];
+            let pltname = &binary.dynstrtab[pltsym.st_name];
+            if pltname == name {
+                debug!("Import match {}: {:#x} {:?}", name, reloc.r_offset, pltsym);
+                proj.imports.insert(reloc.r_offset as u64, name.to_string());
+                return true;
+            }
+        }
+        false
+    };
+
+    // add dynamic symbol information (non-strippable)
+    for sym in &binary.dynsyms {
+        add_sym(&mut prog, sym, &binary.dynstrtab);
+        if sym.is_function () {
+            let name = &binary.dynstrtab[sym.st_name];
+            if !resolve_import_address(&mut proj, &binary.pltrelocs, name) {
+                if !resolve_import_address(&mut proj, &binary.dynrelas, name) {
+                    resolve_import_address(&mut proj, &binary.dynrels,  name);
```

Comment:
I have one last question: Why didn't you define the method `resolve_import_address ` on `Project` itself?